### PR TITLE
Add coverage reporting for JavaScript and TypeScript files

### PR DESCRIPTION
**Pull request by murgatroid99**
This also has the test script run the tests in `test/api`, including the interop sanity test.

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -39,6 +39,16 @@ npm install --unsafe-perm
 
 mkdir -p reports
 export JOBS=8
+export JUNIT_REPORT_PATH="reports/node$version/"
+export JUNIT_REPORT_STACK=1
+
+gsutil cp gs://grpc-testing-secrets/coveralls_credentials/grpc-node.rc .
```
Comment (nicolasnoble) [inline run-tests.sh]: This can actually only work from Linux...

File: run-tests.sh [hunk 2]
```diff
@@ -39,6 +39,16 @@ npm install --unsafe-perm
...
+set +x
+. grpc-node.rc
```
Comment (nicolasnoble) [inline run-tests.sh]: Something seems wrong sourcing from the CWD. Maybe let's try /tmp ?

File: run-tests.sh (hunk 1 above)
Comment (murgatroid99) [reply run-tests.sh]: How would you feel about separating the Linux and Mac test scripts?

Comment (nicolasnoble) [reply run-tests.sh]: Not sure - it seems to be "pick your poison" to me. I don't like to burden the script with various platform-specific checks, but I also don't like the idea of duplicating code.

Note that we're already doing SOME platform-specific work in some of these scripts (https://github.com/grpc/grpc-node/blob/master/tools/release/kokoro.sh#L39) so maybe adding OS-specific code is a slightly less bitter pill to swallow ?

But I don't have any strong opinion one way or another.

**Comment by nicolasnoble** [comment]
Did TypeScript break us again ?

**Comment by jtattermusch** [comment]
Looks like this has broken   grpc/grpc tests on master.

https://source.cloud.google.com/results/invocations/707c0572-1a9c-495e-aabe-81a92504fd21/targets/github%2Fgrpc/tests
```
+ gsutil cp gs://grpc-testing-secrets/coveralls_credentials/grpc-node.rc /tmp
./run-tests.sh: 48: ./run-tests.sh: gsutil: not found
```

**Pull request merged**